internal/identity/mfa/mfa.go

```diff
@@ -0,0 +1,617 @@
+	return getSchemaResource(s, config, mustAddCommonSchema, mustAddCommonMFASchema)
+}
+
+func getSchemaResource(s map[string]*schema.Schema, config *contextFuncConfig, addFuncs ...addSchemaFunc) *schema.Resource {
+	m := map[string]*schema.Schema{}
+	for k, v := range s {
+		m[k] = v
+	}
+
+	r := &schema.Resource{
+		Schema:        m,
+		CreateContext: GetCreateContextFunc(config),
+		UpdateContext: GetUpdateContextFunc(config),
+		ReadContext:   GetReadContextFunc(config),
+		DeleteContext: GetDeleteContextFunc(config),
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+	}
+
+	if len(addFuncs) == 0 {
+		addFuncs = []addSchemaFunc{
+			mustAddCommonSchema,
+		}
+	}
+
+	for _, f := range addFuncs {
+		r = f(r)
+	}
+
+	config.m = r.Schema
+
+	return r
+}
+
+// GetCreateContextFunc for a contextFuncConfig.
+// The return function supports the path types: PathTypeName, and PathTypeMethodID
+func GetCreateContextFunc(config *contextFuncConfig) schema.CreateContextFunc {
+	return func(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+		config.Lock()
+		defer config.Unlock()
+
+		c, dg := provider.GetClientDiag(d, meta)
+		if dg != nil {
+			return dg
+		}
+
+		idField, err := config.IDField()
+		if err != nil {
+			return diag.FromErr(err)
+		}
+
+		var path string
+		if config.IsIDFromResponse() {
+			p, err := config.GetRequestPathBase()
+			if err != nil {
+				return diag.FromErr(err)
+			}
+			path = p
+		} else {
+			p, err := config.GetRequestPathWithID(d.Get(idField).(string))
+			if err != nil {
+				return diag.FromErr(err)
+			}
+			path = p
+		}
+
+		resp, err := c.Logical().Write(path, config.GetRequestData(d))
+		if err != nil {
+			return diag.FromErr(err)
+		}
+
+		var rid string
+		if config.IsIDFromResponse() {
+			id, err := config.GetIDFromResponse(resp)
+			if err != nil {
+				return diag.FromErr(err)
+			}
+
+			rid = id
+		} else {
+			id, err := config.GetIDFromResourceData(d)
+			if err != nil {
+				return diag.FromErr(err)
+			}
+
+			rid = id
+		}
+
+		d.SetId(rid)
+
+		return GetReadContextFunc(config)(ctx, d, meta)
+	}
+}
+
+// GetUpdateContextFunc for a contextFuncConfig.
+// The return function supports the path types: PathTypeName, and PathTypeMethodID
+func GetUpdateContextFunc(config *contextFuncConfig) schema.UpdateContextFunc {
+	return func(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+		config.Lock()
+		defer config.Unlock()
+
+		c, dg := provider.GetClientDiag(d, meta)
+		if dg != nil {
+			return dg
+		}
+
+		id := d.Id()
+		if id == "" {
+			return diag.FromErr(fmt.Errorf("resource ID is empty"))
+		}
+
+		path, err := config.GetRequestPathWithID(id)
+		if err != nil {
+			return diag.FromErr(err)
+		}
+
+		// login MFA does not support partial updates unfortunately,
+		// so we update() becomes very similar to create.
```

Comment:
nit
```suggestion
		// so update() becomes very similar to create.
```

Reply:
Fixed in 75bc3995bb942b958c52cd71e3b23c1eed9a9d8e